src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -306,22 +300,89 @@ ucs_status_t uct_ib_mlx5_devx_modify_qp_state(uct_ib_mlx5_qp_t *qp,
 
 void uct_ib_mlx5_devx_destroy_qp(uct_ib_mlx5_md_t *md, uct_ib_mlx5_qp_t *qp)
 {
-    int ret = mlx5dv_devx_obj_destroy(qp->devx.obj);
-    if (ret) {
-        ucs_error("mlx5dv_devx_obj_destroy(QP) failed: %m");
-    }
+    uct_ib_mlx5_devx_obj_destroy(qp->devx.obj, "QP");
     uct_ib_mlx5_put_dbrec(qp->devx.dbrec);
     uct_ib_mlx5_md_buf_free(md, qp->devx.wq_buf, &qp->devx.mem);
 }
 
+ucs_status_t uct_ib_mlx5_devx_obj_modify(struct mlx5dv_devx_obj *obj,
+                                         const void *in, size_t inlen,
+                                         void *out, size_t outlen,
+                                         char *msg_arg)
+{
+    int ret           = mlx5dv_devx_obj_modify(obj, in, inlen, out, outlen);
+    unsigned syndrome = UCT_IB_MLX5DV_GET(general_obj_out_cmd_hdr, out,
+                                          syndrome);
+
+    if (ret) {
+        ucs_error("mlx5dv_devx_obj_modify(%s) failed, syndrome %x: %m",
+                   msg_arg, syndrome);
+        return UCS_ERR_IO_ERROR;
+    }
+
+    return UCS_OK;
+}
+
+struct mlx5dv_devx_obj*
+uct_ib_mlx5_devx_obj_create(struct ibv_context *context,
+                            const void *in, size_t inlen,
+                            void *out, size_t outlen,
+                            char *msg_arg)
+{
+    struct mlx5dv_devx_obj *obj = mlx5dv_devx_obj_create(context, in, inlen,
+                                                         out, outlen);
+    unsigned syndrome           = UCT_IB_MLX5DV_GET(general_obj_out_cmd_hdr,
+                                                    out, syndrome);
+
+    if (obj == NULL) {
+        ucs_error("mlx5dv_devx_obj_create(%s) failed, syndrome %x: %m",
```

Comment:
print the device name, may need to pass uct_ib_device_t or uct_ib_iface_t
syndrome 0x%x